### DISCUSSION TITLE
Expect beginner-friendly installation and configuration guides or video tutorials

### DISCUSSION THREAD

**Discussion by jinze-liu** [BLAST-WarpX/warpx#4813] [Installation]
Dear WarpX Maintenance Team,

First and foremost, I would like to express my gratitude for your hard work in developing and maintaining WarpX. It is an incredibly powerful and useful tool that has made a significant impact in the field of scientific research.

As a new member of the WarpX community, I have encountered some challenges while trying to install and configure WarpX using Conda. Although the documentation provides some useful information, I found that for beginners like myself, some steps are somewhat difficult to understand, especially those involving environment setup and dependency management.

I believe that a detailed installation and configuration guide for beginners, particularly if accompanied by a video tutorial, would greatly assist new users in getting started with WarpX more quickly. Video tutorials could provide a more intuitive demonstration of the entire process, including common pitfalls and how to resolve potential issues. This would not only enhance the entry experience for new users but also reduce the time community and maintainers spend on addressing installation and configuration-related questions.

I understand that this may require additional time and resources, but I believe it would be a worthwhile investment that could further expand the user base of WarpX and promote community growth. If possible, I am also willing to assist in the creation of the tutorial, whether it be through providing feedback, testing the clarity of the guide, or other means.

Thank you again for your contributions to WarpX and the scientific community at large. I look forward to your response and hope to see more new users joining the WarpX community smoothly.

Best wishes,
jinze.

**Comment by ax3l** [comment]
Dear Jinze,

Thank you for the kind words and for taking the time to share your ideas.

> As a new member of the WarpX community, I have encountered some challenges while trying to install and configure WarpX using Conda. Although the documentation provides some useful information, I found that for beginners like myself, some steps are somewhat difficult to understand, especially those involving environment setup and dependency management.
>
> I believe that a detailed installation and configuration guide for beginners [...] would greatly assist new users in getting started with WarpX more quickly.

That is a valuable experience to share. Can you please provide more details?

We try in fact to write our documentation in a way that if you _copy-paste_ what is written, you should (for desktop and HPC instructions) arrive at a usable installation.

At what point of which page of the manual did you encounter which exact issues? Please post your exact steps and exact output/errors you encountered (or steps we forgot to document that you found missing).

> Video tutorials.

There are a couple of angles to this that I would like to understand better to know what you are asking for.

Generally, we can sit down, screen record how we copy-paste the instructions from the manual to a terminal and it will be the same as the manual - it will just outdate faster and is not as reproducibile (or automatable) as following the manual.

Another downside I see: a video tutorial is still not _interactive_: if a user encounters a particular issue, e.g., because something is special on their system, the most interactive way to provide feedback is to report an issue or discussion on GitHub, so we can triage, assign and solve it together.
(Contrary, adding comments under a YouTube video is very hard to track or provide support for when we need technical details such as error logs, system details, and need to share scripts.)

Are you maybe referring to the _graphical installer_ of Conda, where you click and point and enter packages manually instead of running a reproducible script in the terminal? That workflow indeed benefits from a video tutorial.

> I understand that this may require additional time and resources, but I believe it would be a worthwhile investment that could further expand the user base of WarpX and promote community growth. If possible, I am also willing to assist in the creation of the tutorial, whether it be through providing feedback, testing the clarity of the guide, or other means.

That is a great offer and your help is definitely welcome! :)
Let's identify what is needed first and then devise a plan how we can address onboarding even better with such short video tutorials.

Best,
Axel

**Reply by jinze-liu** [reply]
Thank you for your reply. I succeeded in installing WarpX via conda on my Linux system and have been following the example provided in the Laser-Wakefield Acceleration of Electrons documentation. I executed the Python script as described and it seems to have run successfully, generating two output directories in my designated folder.

However, when I attempt to visualize the simulation results using plot_3d.py as per the documentation, I encounter difficulties. Initially, I received a "command not found" error when trying to execute plot_3d.py. After  I then encountered an ImportError: No module named 'matplotlib'.

Despite attempting to reinstall matplotlib to resolve this issue, the problem persists. Visualization is a crucial step for me, and I'm currently facing challenges with it. Could you provide any guidance or suggestions to help resolve these visualization issues?

Thank you for your support

**Reply by ax3l** [reply]
Thanks for the details!

If you want to visualize results via Python, you need to install the needed packages. The [visualization script](https://warpx.readthedocs.io/en/latest/usage/examples/lwfa/README.html#visualize) you just needs matplotlib. [Install it in the same conda environment as WarpX with either conda or pip.](https://matplotlib.org/stable/users/installing/index.html)
